### PR TITLE
First commits

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request implements a Terraform module for installing into an account an AWS Lambda function that will disable users' inactive console and programmatic access.  The Lambda function is run on a schedule via AWS EventBridge.

See also:
- cisagov/provisionaccount-role-tf-module/pull/45
- cisagov/disable-inactive-iam-users-lambda#1
- cisagov/cool-accounts#147

## 💭 Motivation and context ##

We have a POAM about disabling inactive users in IAM.  Partially resolves cisagov/cool-system-internal#143.

## 🧪 Testing ##

I applied this module to our COOL Users account without incident.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,28 +42,40 @@ No modules.
...
 ## Inputs ##
 
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
-| ami\_owner\_account\_id | The ID of the AWS account that owns the Example AMI, or "self" if the AMI is owned by the same account as the provisioner. | `string` | `"self"` | no |
-| aws\_availability\_zone | The AWS availability zone to deploy into (e.g. a, b, c, etc.). | `string` | `"a"` | no |
 | aws\_region | The AWS region to deploy into (e.g. us-east-1). | `string` | `"us-east-1"` | no |
-| subnet\_id | The ID of the AWS subnet to deploy into (e.g. subnet-0123456789abcdef0). | `string` | n/a | yes |
+| expiration\_days | A strictly positive integer denoting the number of days after which an IAM user's access is considered inactive if unused. | `number` | `45` | no |
+| lambda\_function\_description | The description of the Lambda function. | `string` | `"Lambda function to disable inactive IAM users."` | no |
```
Comment (dav3r) [inline README.md]: I think we usually use the same verbiage in the description as we do in the name:
```suggestion
| lambda\_function\_description | The description of the Lambda function that will disable inactive IAM users. | `string` | `"Lambda function to disable inactive IAM users."` | no |
```

**Review by dv4harr10 — APPROVED** [review]

File: variables.tf [hunk 2]
```diff
@@ -4,30 +4,67 @@
...
+
+  validation {
+    condition     = alltrue([floor(var.expiration_days) == var.expiration_days, var.expiration_days > 0])
+    error_message = "expiration_days must be a positive integer greater than zero."
+  }
+}
+
+variable "lambda_function_description" {
+  default     = "Lambda function to disable inactive IAM users."
+  description = "The description of the Lambda function."
```
Comment (dav3r) [inline variables.tf]: Corresponding to a similar suggestion above:
```suggestion
  description = "The description of the Lambda function that will disable inactive IAM users."
```

File: lambda.tf [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+# ------------------------------------------------------------------------------
+
+resource "aws_lambda_function" "disable_inactive_iam_users" {
+  description      = var.lambda_function_description
+  filename         = var.lambda_zip_filename
+  function_name    = var.lambda_function_name
+  handler          = "lambda_handler.handler"
+  memory_size      = 128
+  role             = aws_iam_role.lambdaexecution_role.arn
+  runtime          = "python3.9"
```
Comment (dav3r) [inline lambda.tf]: Is there some reason why we wouldn't want to update this to run with `python3.11` or `python3.12`?

https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

**Review by dav3r — APPROVED** [review]
I noted a couple of thangs, but nothing earth-shattering. 👍

File: lambda.tf (hunk 3 above)
Comment (jsf9k) [reply lambda.tf]: I was waiting for our skeleton to support it.  It would be easy enough to add, though.

Comment (dav3r) [reply lambda.tf]: In this case, I'm fine with getting ahead of the skeleton.  Let us know what you decide and I'll resolve this conversation either way.

Comment (jsf9k) [reply lambda.tf]: Our Terraform provider constraints only allow us to support Python 3.10 or earlier, so I will go with that.

Comment (jsf9k) [reply lambda.tf]: Please see commit 3ddb152ad5f212d25fae4204647e4327a78df46b and commit cisagov/disable-inactive-iam-users-lambda@311f8b8d2ea7997042970e4b1d025173f49ac28a.

**Review by dav3r — APPROVED** [review]
👍 👍

File: lambda.tf (hunk 3 above)
Comment (dav3r) [reply lambda.tf]: Also, we need to move to a Terraform provider that supports Python 3.12 (required for what we are building over in https://github.com/cisagov/cyhy-kevsync-lambda/tree/first-commits-felddy).

**Pull request merged**